certmanager/checkupdate.py

```
@@ -0,0 +1,271 @@
+            if not response_data or not isinstance(response_data, dict):
+                print(f"[{PARTNER_ID}] Invalid or empty response.")
+                continue
+
+            cert_info = response_data.get('response')
+            CERTIFICATE_DATA = cert_info.get('certificateData') if cert_info else None
+
+            if not CERTIFICATE_DATA:
+                print(f"[{PARTNER_ID}] Certificate data not found.")
+                continue
+
+            expiration_date = os.popen(f"echo '{CERTIFICATE_DATA}' | openssl x509 -noout -enddate").read().split('=')[1].strip()
```

Ruff (0.16.3)

[error] Starting a process with a shell, possible injection detected

(S605)